### PR TITLE
Add sideload node and service

### PR DESCRIPTION
**Pull request by nathanielc**
Adds a sideload node that acts similar to the default node but allows the data to be loaded from external sources.

TODO
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Trigger source update over API or signal
- [x] Check only files below the source can be used.
- [x] Cache source objects so they are reused per unique source
- [x] Get feedback on `{tag}` syntax.
- [x] Figure out what to do about conflicting ToRow usage.
- [x] Update API docs
- [ ] Update CLI to trigger a sideload reload (Decided not to do this)

### CODE REVIEW

File: sideload.go [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+	s          *pipeline.SideloadNode
+	source     sideload.Source
+	orderTmpls []orderTmpl
+
+	order []string
+
+	bufferPool *bufpool.Pool
+}
+
+// Create a new  SideloadNode which shifts points and batches in time.
```
Comment (desa) [inline sideload.go]: > which shifts points and batches in time.

Is this right?

File: sideload.go [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+func newSideloadNode(et *ExecutingTask, n *pipeline.SideloadNode, d NodeDiagnostic) (*SideloadNode, error) {
+	sn := &SideloadNode{
+		node:       node{Node: n, et: et, diag: d},
+		s:          n,
+		bufferPool: bufpool.New(),
+		order:      make([]string, len(n.OrderList)),
+		orderTmpls: make([]orderTmpl, len(n.OrderList)),
+	}
+	u, err := url.Parse(n.Source)
```
Comment (desa) [inline sideload.go]: Should lines 40-49 go into the et.tm.SideloadService.Source method?

File: sideload.go [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+func (n *SideloadNode) doSideload(p edge.FieldsTagsTimeSetter) {
+	for i, o := range n.orderTmpls {
+		p, err := o.Path(p.Tags())
+		if err != nil {
+			n.diag.Error("failed to evaluate order template", err, keyvalue.KV("order", o.raw))
+			return
+		}
+		n.order[i] = p
+	}
+	if len(n.s.Fields) > 0 {
```
Comment (desa) [inline sideload.go]: Sucks that the logic for fields and tags can't be pulled into a common method without adding reflection.

Comment (desa) [reply sideload.go]: :(

File: sideload.go (hunk 1 above)
Comment (nathanielc) [reply sideload.go]: Nope,:)

File: sideload.go (hunk 2 above)
Comment (nathanielc) [reply sideload.go]: Yeah, that would make more sense.

**Pull request merged**